python/ray/tune/trial.py

```diff
@@ -232,7 +232,11 @@ def node_ip(self):
 
     @property
     def checkpoint(self):
-        return self.checkpoint_manager.newest_checkpoint
+        if self.status == Trial.PAUSED:
+            assert self.checkpoint_manager.newest_memory_checkpoint.value
+            return self.checkpoint_manager.newest_memory_checkpoint
+        else:
+            return self.checkpoint_manager.newest_checkpoint
```

Comment:
This is a bit unintuitive - why not always return a Checkpoint object?

Reply:
can you clarify what you mean - it does return a checkpoint obj always

if its paused it returns the memory checkpoint (which it should because PAUSED implies we want to resume from memory). if its any other status then return the most recent persistent checkpoint

Reply:
Nevermind; misread.